src/inmanta/server/services/paramservice.py

```diff
@@ -200,54 +217,69 @@ async def _update_param(
 
-    @handle(methods.set_param, param_id="id", env="tid")
+    @handle(methods.set_param, name="id", env="tid")
     async def set_param(
         self,
         env: data.Environment,
-        param_id: str,
+        name: str,
         source: ParameterSource,
         value: str,
-        resource_id: str,
+        resource_id: Optional[str],
         metadata: JsonType,
         recompile: bool,
+        expires: bool = False,
     ) -> Apireturn:
-        result = await self._update_param(env, param_id, value, source, resource_id, metadata, recompile)
+        result = await self._update_param(env, name, value, source, resource_id, metadata, recompile, expires)
         warnings = None
         if result:
             compile_metadata = {
                 "message": "Recompile model because one or more parameters were updated",
                 "type": "param",
-                "params": [(param_id, resource_id)],
+                "params": [(name, resource_id)],
             }
             warnings = await self.server_slice._async_recompile(env, False, metadata=compile_metadata)
 
         if resource_id is None:
             resource_id = ""
 
-        params = await data.Parameter.get_list(environment=env.id, name=param_id, resource_id=resource_id)
+        params = await data.Parameter.get_list(environment=env.id, name=name, resource_id=resource_id)
 
         return attach_warnings(200, {"parameter": params[0]}, warnings)
 
     @handle(methods.set_parameters, env="tid")
     async def set_parameters(self, env: data.Environment, parameters: list[dict[str, Any]]) -> Apireturn:
         recompile = False
 
+        updating_facts: bool = False
+        updating_parameters: bool = False
+        parameters_and_or_facts: str = "parameters"
+
         params: list[tuple[str, ResourceIdStr]] = []
         for param in parameters:
             name: str = param["id"]
             source = param["source"]
             value = param["value"] if "value" in param else None
             resource_id: ResourceIdStr = param["resource_id"] if "resource_id" in param else None
             metadata = param["metadata"] if "metadata" in param else None
-            expires = param["expires"] if "expires" in param else True
+            expires = param["expires"] if "expires" in param else None
 
-            result = await self._update_param(env, name, value, source, resource_id, metadata, expires)
+            if resource_id:
+                updating_facts = True
+            else:
+                updating_parameters = True
+
+            result = await self._update_param(env, name, value, source, resource_id, metadata, expires=expires)
             if result:
                 recompile = True
                 params.append((name, resource_id))
 
+        if updating_facts:
+            parameters_and_or_facts = "facts"
+        if updating_parameters and updating_facts:
+            parameters_and_or_facts = "parameters and facts"
```

Comment:
Aren't we missing the third option here where only parameters were updated?

Reply:
If no facts were set then the default value for the `parameters_and_or_facts` variable will be used ( "parameters" ).
If this is not clear I can try to make this more explicit

Reply:
Correct. I missed that.